client/rest/src/test/java/org/elasticsearch/client/RestClientBuilderTests.java

```diff
@@ -103,41 +104,82 @@ public void testBuild() throws IOException {
 
-        int numNodes = RandomInts.randomIntBetween(getRandom(), 1, 5);
+        int numNodes = randomIntBetween(1, 5);
         HttpHost[] hosts = new HttpHost[numNodes];
         for (int i = 0; i < numNodes; i++) {
             hosts[i] = new HttpHost("localhost", 9200 + i);
         }
         RestClientBuilder builder = RestClient.builder(hosts);
-        if (getRandom().nextBoolean()) {
+        if (randomBoolean()) {
             builder.setHttpClientConfigCallback(new RestClientBuilder.HttpClientConfigCallback() {
                 @Override
                 public HttpAsyncClientBuilder customizeHttpClient(HttpAsyncClientBuilder httpClientBuilder) {
                     return httpClientBuilder;
                 }
             });
         }
-        if (getRandom().nextBoolean()) {
+        if (randomBoolean()) {
             builder.setRequestConfigCallback(new RestClientBuilder.RequestConfigCallback() {
                 @Override
                 public RequestConfig.Builder customizeRequestConfig(RequestConfig.Builder requestConfigBuilder) {
                     return requestConfigBuilder;
                 }
             });
         }
-        if (getRandom().nextBoolean()) {
-            int numHeaders = RandomInts.randomIntBetween(getRandom(), 1, 5);
+        if (randomBoolean()) {
+            int numHeaders = randomIntBetween(1, 5);
             Header[] headers = new Header[numHeaders];
             for (int i = 0; i < numHeaders; i++) {
                 headers[i] = new BasicHeader("header" + i, "value");
             }
             builder.setDefaultHeaders(headers);
         }
-        if (getRandom().nextBoolean()) {
-            builder.setMaxRetryTimeoutMillis(RandomInts.randomIntBetween(getRandom(), 1, Integer.MAX_VALUE));
+        if (randomBoolean()) {
+            builder.setMaxRetryTimeoutMillis(randomIntBetween(1, Integer.MAX_VALUE));
+        }
+        if (randomBoolean()) {
+            String basePath = randomAsciiOfLengthBetween(2, 5);
+
+            while (basePath.length() < 20 && randomBoolean()) {
+                basePath += "/" + randomAsciiOfLengthBetween(3, 6);
+            }
+
+            builder.setBasePath(basePath);
         }
         try (RestClient restClient = builder.build()) {
             assertNotNull(restClient);
         }
     }
+
+    public void testSetBasePathNull() {
+        try {
+            RestClient.builder(new HttpHost("localhost", 9200)).setBasePath(null);
+        } catch (final NullPointerException e) {
+            assertEquals("basePath must not be null", e.getMessage());
+        }
+    }
+
+    public void testSetBasePathEmpty() {
+        assertSetBasePathThrows("  /  ");
+        assertSetBasePathThrows(" / ");
+        assertSetBasePathThrows("/");
+        assertSetBasePathThrows(" ");
+        assertSetBasePathThrows("");
+    }
+
+    public void testSetBasePathMalformed() {
+        assertSetBasePathThrows("//");
+        assertSetBasePathThrows("base/path//");
```

Comment:
use expectThrows instead and drop the added method as each assert will be a single line without it too?